langchain4j-qianfan/src/test/java/dev/langchain4j/model/qianfan/QianfanStreamingChatModelIT.java

```diff
@@ -0,0 +1,224 @@
+package dev.langchain4j.model.qianfan;
+
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ToolExecutionResultMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.model.StreamingResponseHandler;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.model.output.TokenUsage;
+import org.junit.jupiter.api.Test;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import static dev.langchain4j.agent.tool.JsonSchemaProperty.INTEGER;
+import static dev.langchain4j.data.message.ToolExecutionResultMessage.from;
+import static dev.langchain4j.data.message.UserMessage.userMessage;
+import static dev.langchain4j.model.output.FinishReason.STOP;
+import static dev.langchain4j.model.output.FinishReason.TOOL_EXECUTION;
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+class QianfanStreamingChatModelIT {
+
+    //see your api key and secret key here: https://console.bce.baidu.com/qianfan/ais/console/applicationConsole/application
+    private String apiKey ="your api key";
+    private String secretKey ="your secret key";
+    QianfanStreamingChatModel model = QianfanStreamingChatModel.builder().modelName("ERNIE-Bot 4.0").temperature(0.7).topP(1.0f)
+            .apiKey(apiKey)
+            .secretKey(secretKey)
+            .build();
+    ToolSpecification calculator = ToolSpecification.builder()
+            .name("calculator")
+            .description("returns a sum of two numbers")
+            .addParameter("first", INTEGER)
+            .addParameter("second", INTEGER)
+            .build();
+
+
+
+  @Test
+    void should_stream_answer() throws ExecutionException, InterruptedException, TimeoutException {
+
+
+        CompletableFuture<Response<AiMessage>> futureResponse = new CompletableFuture<>();
+
+        model.generate("中国首都是哪里", new StreamingResponseHandler<AiMessage>() {
```

Comment:
please use `TestStreamingResponseHandler` instead of creating anonymous `StreamingResponseHandler` every time (in all tests)
For this you will need to add
```
        <dependency>
            <groupId>dev.langchain4j</groupId>
            <artifactId>langchain4j-core</artifactId>
            <classifier>tests</classifier>
            <type>test-jar</type>
            <scope>test</scope>
        </dependency>
```